### ISSUE TITLE
Constraint failed on primary key when using bulk insert

### ISSUE THREAD

**Issue by claunia** [borisdj/EFCore.BulkExtensions#380]
Exception:
```
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteNonQuery()
   at EFCore.BulkExtensions.SqlBulkOperation.Merge[T](DbContext context, Type type, IList`1 entities, TableInfo tableInfo, OperationType operationType, Action`1 progress)
   at EFCore.BulkExtensions.SqlBulkOperation.Merge[T](DbContext context, IList`1 entities, TableInfo tableInfo, OperationType operationType, Action`1 progress)
   at EFCore.BulkExtensions.DbContextBulkTransaction.Execute[T](DbContext context, IList`1 entities, OperationType operationType, BulkConfig bulkConfig, Action`1 progress)
   at EFCore.BulkExtensions.DbContextBulkExtensions.BulkInsert[T](DbContext context, IList`1 entities, Action`1 bulkAction, Action`1 progress)
   at RomRepoMgr.Core.Workers.DatImporter.Import() in /home/claunia/Development/RomRepoMgr/RomRepoMgr.Core/Workers/DatImporter.cs:line 187
```

Model:
```csharp
    public class Machine : BaseModel<ulong>
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public virtual RomSet RomSet { get;                        set; }
        public virtual ICollection<FileByMachine>  Files    { get; set; }
        public virtual ICollection<DiskByMachine>  Disks    { get; set; }
        public virtual ICollection<MediaByMachine> Medias   { get; set; }
        public         long                        RomSetId { get; set; }
    }
```

Base model:
```csharp
    public abstract class BaseModel<TKey>
    {
        public TKey Id { get; set; }

        public DateTime CreatedOn { get; set; }
        public DateTime UpdatedOn { get; set; }
    }
```

Context:
```csharp
        public DbSet<Machine>        Machines         { get; set; }

            modelBuilder.Entity<Machine>(entity =>
            {
                entity.HasIndex(e => e.Name);

                entity.HasOne(e => e.RomSet).WithMany(e => e.Machines).OnDelete(DeleteBehavior.Cascade);
            });
```

Snapshot:
```csharp
modelBuilder.Entity("RomRepoMgr.Database.Models.Machine", b =>
            {
                b.Property<ulong>("Id").ValueGeneratedOnAdd().HasColumnType("INTEGER");

                b.Property<DateTime>("CreatedOn").HasColumnType("TEXT");

                b.Property<string>("Name").IsRequired().HasColumnType("TEXT");

                b.Property<long>("RomSetId").HasColumnType("INTEGER");

                b.Property<DateTime>("UpdatedOn").HasColumnType("TEXT");

                b.HasKey("Id");

                b.HasIndex("Name");

                b.HasIndex("RomSetId");

                b.ToTable("Machines");
            });
```

Insertion code:
```csharp
                Dictionary<string, Machine> machines = new Dictionary<string, Machine>();

                foreach(string name in machineNames)
                {
                    var machine = new Machine
                    {
                        Name      = name,
                        RomSetId  = romSet.Id, // 1
                        CreatedOn = DateTime.UtcNow,
                        UpdatedOn = DateTime.UtcNow
                    };

                    machines[name] = machine;
                }

                ctx.BulkInsert(machines.Values.ToList(), b => b.SetOutputIdentity=true);
```

**Comment by claunia** [comment]
Checking at your source in:
https://github.com/borisdj/EFCore.BulkExtensions/blob/977072179e85f6b4e4add7e75048a382ab5378e8/EFCore.BulkExtensions/SqlBulkOperation.cs#L122

I can see you're sending:
```sql
INSERT INTO [Machines] ([Id], [CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES (0, "", "", 1, "");
```

Then again the same. Dunno SQL Server but if you send that in SQLite it will set the Id column:
```sql
sqlite> INSERT INTO [Machines] ([Id], [CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES (0, "", "", 1, "");
sqlite> select * from machines;
0||||1
sqlite> INSERT INTO [Machines] ([Id], [CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES (0, "", "", 1, "");
Error: UNIQUE constraint failed: Machines.Id
```

You need to skip autogenerated Id column like this:
```sql
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
sqlite> INSERT INTO [Machines] ([CreatedOn], [Name], [RomSetId], [UpdatedOn]) VALUES ("", "", 1, "");
```

The code seems to be properly detecting that is the primary key.

However the check in:
https://github.com/borisdj/EFCore.BulkExtensions/blob/977072179e85f6b4e4add7e75048a382ab5378e8/EFCore.BulkExtensions/TableInfo.cs#L147

seems incorrect.

Indeed changing all the primary key types to `int` fixes the problem, so that line should be changed to:
```csharp
  IdentityColumnName = allProperties.SingleOrDefault(a => a.IsPrimaryKey() && (a.ClrType.Name.StartsWith("Byte") || a.ClrType.Name.StartsWith("SByte") || a.ClrType.Name.StartsWith("Int") || a.ClrType.Name.StartsWith("UInt")) && !a.ClrType.Name.EndsWith("[]") && a.ValueGenerated == ValueGenerated.OnAdd)?.Name; // ValueGenerated equals OnAdd even for nonIdentity column like Guid so we only type int as second condition
```
to fix it.

**Comment by michal-ciechan** [comment]
Also need to add `Int64` / `UInt64` ("long"), just got this breaking when upgrading this package. Will have to roll back for now.

**Comment by borisdj** [comment]
Fixed with v3.2.6
